File: service/DTO/TutorialDTO.cs
namespace PaintLearning.Service.DTO;

public class TutorialDTO
{
    public string Id { get; set; } = string.Empty;
    public List<TutorialStepDTO> Steps { get; set; } = new List<TutorialStepDTO>();
    public List<string> Materials { get; set; } = new List<string>();
    public List<string> Tips { get; set; } = new List<string>();
}

public class TutorialStepDTO
{
    public int Order { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public List<string> KeyPoints { get; set; } = new List<string>();
    public string? RegionHint { get; set; }
}

File: backend/src/main/java/com/paintlearning/app/dto/response/TutorialDTO.java
package com.paintlearning.app.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TutorialDTO {
    private String id;
    private List<TutorialStepDTO> steps;
    private List<String> materials;
    private List<String> tips;
}